src/app-layer-smb.c

```diff
@@ -230,95 +38,11 @@ static SuricataFileContext sfc = { &sbcfg };
-            /* if we have no config, we enable the default port 445 */
-            if (!have_cfg) {
-                SCLogConfig("no SMB TCP config found, enabling SMB detection "
-                            "on port 445.");
-                AppLayerProtoDetectPPRegister(IPPROTO_TCP, "445", ALPROTO_SMB, 0,
-                        MIN_REC_SIZE, STREAM_TOSERVER, SMBTCPProbe,
-                        SMBTCPProbe);
-            }
-        }
-    } else {
-        SCLogConfig("Protocol detection and parser disabled for %s protocol.",
-                  proto_name);
-        return;
-    }
-
-    if (AppLayerParserConfParserEnabled("tcp", proto_name)) {
-        AppLayerParserRegisterParser(IPPROTO_TCP, ALPROTO_SMB, STREAM_TOSERVER,
-                SMBTCPParseRequest);
-        AppLayerParserRegisterParser(IPPROTO_TCP , ALPROTO_SMB, STREAM_TOCLIENT,
-                SMBTCPParseResponse);
-        AppLayerParserRegisterStateFuncs(IPPROTO_TCP, ALPROTO_SMB,
-                rs_smb_state_new, rs_smb_state_free);
-        AppLayerParserRegisterTxFreeFunc(IPPROTO_TCP, ALPROTO_SMB,
-                SMBStateTransactionFree);
-
-        AppLayerParserRegisterGetEventsFunc(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetEvents);
-        AppLayerParserRegisterGetEventInfo(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetEventInfo);
-        AppLayerParserRegisterGetEventInfoById(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetEventInfoById);
-
-        AppLayerParserRegisterDetectStateFuncs(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetTxDetectState, SMBSetTxDetectState);
-        AppLayerParserRegisterGetTx(IPPROTO_TCP, ALPROTO_SMB, SMBGetTx);
-        AppLayerParserRegisterGetTxIterator(IPPROTO_TCP, ALPROTO_SMB, SMBGetTxIterator);
-        AppLayerParserRegisterGetTxCnt(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetTxCnt);
-        AppLayerParserRegisterGetStateProgressFunc(IPPROTO_TCP, ALPROTO_SMB,
-                SMBGetAlstateProgress);
-        AppLayerParserRegisterStateProgressCompletionStatus(ALPROTO_SMB, 1, 1);
-        AppLayerParserRegisterTruncateFunc(IPPROTO_TCP, ALPROTO_SMB,
-                                          SMBStateTruncate);
-        AppLayerParserRegisterGetFilesFunc(IPPROTO_TCP, ALPROTO_SMB, SMBGetFiles);
-
-        AppLayerParserRegisterTxDataFunc(IPPROTO_TCP, ALPROTO_SMB, rs_smb_get_tx_data);
-
-        /* This parser accepts gaps. */
-        AppLayerParserRegisterOptionFlags(IPPROTO_TCP, ALPROTO_SMB,
-                APP_LAYER_PARSER_OPT_ACCEPT_GAPS);
-
-        ConfNode *p = ConfGetNode("app-layer.protocols.smb.stream-depth");
-        if (p != NULL) {
-            uint32_t value;
-            if (ParseSizeStringU32(p->val, &value) < 0) {
-                SCLogError(SC_ERR_SMB_CONFIG, "invalid value for stream-depth %s", p->val);
-            } else {
-                stream_depth = value;
-            }
-        }
-        SCLogConfig("SMB stream depth: %u", stream_depth);
```

Comment:
Although there is just TCP registration here, to be sure, is that all we need?

Reply:
Yes we support smb over tcp only.